### PR TITLE
added functions for chadwick data

### PR DESCRIPTION
**Pull request by valdezt**
Switched retrosheet module to pull from the Chadwick Github repo for data. This way, checking for dates and files is (in my opinion) a little easier. This also enables download of single files.

Other files that are now downloadable:
- Stadium data
- Rosters
- Schedules
- Event files

### CODE REVIEW

File: pybaseball/retrosheet.py [hunk 1]
```diff
@@ -21,81 +21,230 @@
...
+        g = Github(GH_TOKEN)
+        try:
+            repo = g.get_repo('chadwickbureau/retrosheet')
+            tree = repo.get_git_tree('master')
+            for t in tree.tree:
+                if t.path == 'rosters':
+                    subtree = t
+
+            rosters = [t.path for t in repo.get_git_tree(subtree.sha).tree]
+            file_name = '{}{}.ROS'.format(team, season)
```
Comment (TylerLiu42) [inline pybaseball/retrosheet.py]: nitpick: Use f-strings?

**Comment by schorrm** [comment]
To clarify -- does this have the ZIP functionality still in place?

**Comment by valdezt** [comment]
Fixed to use f-strings.

You're no longer able to download the ZIP files, but you're able to download the same data from a different source. The function to read/unzip the files still exists in the utils.

File: docs/retrosheet.md [hunk 2]
```diff
@@ -23,6 +23,12 @@ material presented here. All information is subject to corrections
 as additional data are received. We are grateful to anyone who
 discovers discrepancies and we appreciate learning of the details.
 
+## Extra Setup (Optional)
+
+When pulling retrosheet data, the request is checked against the files in the Chadwick Bureau retrosheet Github repository. The Github API has a rate limit of 60 queries per hour for non-registered queries. To register your connection and raise your rate limit to 5000 queries per hour, follow [this guide](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) once you've signed up for a Github account. On a MAC/Linux system, add `GH_TOKEN=<your_token>` to your `.bashrc` file. On Windows, add `GH_TOKEN` to your environment variables via the Start menu.
```
Comment (schorrm) [inline docs/retrosheet.md]: Typo: Mac

File: pybaseball/retrosheet.py [hunk 3]
```diff
@@ -21,81 +21,230 @@
...
+        rosters = [t.path for t in repo.get_git_tree(subtree.sha).tree if str(season) in t.path]
+        if len(rosters) == 0:
+            raise ValueError(f'Rosters not available for {season}')
+    except RateLimitExceededException:
+        warnings.warn(
+            'Github rate limit exceeded. Cannot check if the file you want exists.',
+            UserWarning
+        )
+
+    df_list = [get_roster(team = r[:3], season = season, checked=False) for r in rosters]
```
Comment (schorrm) [inline pybaseball/retrosheet.py]: This doesn't exist. I guess you mean `_roster`?

**Review by schorrm — COMMENTED** [review]
Caught a bug here.
This is a great PR, just fix the one bug and add the new methods to the `__init__.py`.

**Comment by valdezt** [comment]
Made all the changes requested. Also added pygithub to the setup.py requirements.

Thanks for the review. Hoping to build a retrosheet parser for the event logs sometime soon. I've started but it'll be a bit of a long haul I think.

**Comment by schorrm** [comment]
Looks great. Re retrosheet -- I recently found a GitHub repo that has prebuilt databases of the Retrosheet play-by-play, and some people seem to have started play-by-play stuff in Python also (or: can we somehow wrap the C++ parser nicely?)

**Comment by schorrm** [comment]
@valdezt saw that you fixed a typo -- we good to go now on this?

**Comment by valdezt** [comment]
@schorrm yep. I checked the functions work. I think the typo was from before my updates, just saw it needed one less underscore. Should be good to go!

**Comment by valdezt** [comment]
> Looks great. Re retrosheet -- I recently found a GitHub repo that has prebuilt databases of the Retrosheet play-by-play

Could you share this?

> and some people seem to have started play-by-play stuff in Python also (or: can we somehow wrap the C++ parser nicely?)

Yeah I found a [retrosheet repo](https://github.com/calestini/retrosheet) in Python as well as the Python code the [Chadwick Bureau](https://github.com/chadwickbureau/chadwick/tree/master/src/python) has. The Chadwick code is only for Python 2.7. The calestini repo works but I'm not a fan of the output of that package. I could try looking into a wrapper on the C++ code but I'd have to figure out how to make sure it's installed properly during pip installation of this repo.

**Comment by schorrm** [comment]
The prebuilt DB / parquet images may be worth working with.

**Comment by schorrm** [comment]
Maybe better to work with prebuilt DBs or Parquet.
Anyway, great job!

**Comment by bdilday** [comment]
I recently published a Python wrapper of the chadwick c code that might be useful here,
https://pypi.org/project/pychadwick/
https://github.com/bdilday/pychadwick

**Comment by schorrm** [comment]
@bdilday that's really amazing! I would love to hear any ideas that you have for integrating this!

**Comment by schorrm** [comment]
@bdilday want to open an issue for this?

**Pull request merged**